src/dcast.d

```diff
@@ -3182,7 +3182,12 @@ Lagain:
     {
         // Bugzilla 13841, all vector types should have no common types between
         // different vectors, even though their sizes are same.
-        goto Lincompatible;
+        auto tv1 = cast(TypeVector)t1;
+        auto tv2 = cast(TypeVector)t2;
```

Comment:
It is in this case. I had it your way initially, and it made the expression pretty hard to read. Breaking up expressions that do too many things at once can be desirable (and it does not change the code generation).